yarn-project/prover-client/src/orchestrator/block-building-helpers.ts

```diff
@@ -246,14 +274,38 @@ export async function getRootRollupInput(
     0,
   );
 
-  return RootRollupInputs.from({
+  return BlockRootRollupInputs.from({
     previousRollupData,
     l1ToL2Roots,
     newL1ToL2Messages,
     newL1ToL2MessageTreeRootSiblingPath: messageTreeRootSiblingPath,
     startL1ToL2MessageTreeSnapshot: messageTreeSnapshot,
     startArchiveSnapshot,
     newArchiveSiblingPath,
+    // TODO(#7346): Inject previous block hash (required when integrating batch rollup circuits)
+    previousBlockHash: Fr.ZERO,
+    proverId,
+  });
+}
+
+// Builds the inputs for the final root rollup circuit, without making any changes to trees
+// TODO(#7346): Integrate batch rollup circuits and test below
+export function getRootRollupInput(
+  rollupOutputLeft: BlockRootOrBlockMergePublicInputs,
+  rollupProofLeft: RecursiveProof<typeof NESTED_RECURSIVE_PROOF_LENGTH>,
+  verificationKeyLeft: VerificationKeyAsFields,
+  rollupOutputRight: BlockRootOrBlockMergePublicInputs,
+  rollupProofRight: RecursiveProof<typeof NESTED_RECURSIVE_PROOF_LENGTH>,
+  verificationKeyRight: VerificationKeyAsFields,
+  proverId: Fr,
+) {
+  const previousRollupData: RootRollupInputs['previousRollupData'] = [
+    getPreviousRollupBlockDataFromPublicInputs(rollupOutputLeft, rollupProofLeft, verificationKeyLeft),
+    getPreviousRollupBlockDataFromPublicInputs(rollupOutputRight, rollupProofRight, verificationKeyRight),
```

Comment:
Is it possible for the verification keys to be different?
LE: aaahhh.. wonky rollups?

Reply:
Yep exactly! Am leaving it to be flexible in case we want wonkiness from block-root up to root. If we always want a balanced tree (e.g. always 32 block roots per root), this can become one vk.